nucypher/characters/lawful.py

```diff
@@ -163,6 +165,33 @@ def get_policy_pubkey_from_label(self, label: bytes) -> UmbralPublicKey:
         policy_pubkey = alice_delegating_power.get_pubkey_from_label(label)
         return policy_pubkey
 
+    def revoke(self, policy):
+        """
+        Parses the treasure map and revokes arrangements in it.
+        If any arrangements can't be revoked, then the node_id and
+        arrangement_id are added to a list and returned. If a node returns a
+        404 status code, then it will add a tuple of the Revocation and `None`.
+        """
+        try:
+            # Wait for a revocation threshold of nodes to be known ((n - m) + 1)
+            revocation_threshold = ((policy.n - policy.treasure_map.m) + 1)
+            self.block_until_specific_nodes_are_known(
+                    policy.revocation_kit.revokable_addresses,
+                    allow_missing=revocation_threshold)
+        except self.NotEnoughTeachers as e:
+            raise e
+        else:
+            failed_revocations = list()
+            for node_id in policy.revocation_kit.revokable_addresses:
+                ursula = self.known_nodes[node_id]
+                revocation = policy.revocation_kit[node_id]
+                response = self.network_middleware.revoke_arrangement(ursula, revocation)
+                if response.status_code == 404:
+                    failed_revocations.append((revocation, None))
```

Comment:
Can this `None` instead be a constant sorrow?

Reply:
I'm not sure I understand where y'all think things are better as constant sorrow constants. It seems absolutely arbitrary. What is the criteria here?

Reply:
I just have no idea what this `None` indicates when I look at it; You are indicating something specific here, what is it?

Reply:
Ah, I see your point. I have a better idea..

Reply:
Alright, see the latest commit.